gears/carto_gears_api/lib/carto_gears_api/users/users_service.rb

```
@@ -53,6 +53,38 @@ def make_builder(user_id, quota_in_bytes: nil)
         CartoGearsApi::Users::User.from_model(user)
       end
 
+      # Checks the password of a user
+      #
+      # @param user_id [UUID] the user id
+      # @param password [String] password to check
+      # @return [Boolean] true if the password matches with the user, false otherwise
+      #
+      # @raise [Errors::RecordNotFound] if the user could not be found in the database
+      def valid_password?(user_id, password)
+        user = find_user(user_id)
+
+        user.validate_old_password(password)
```

Be careful with this since it will always returns true if the user is configured to sue Google/GitHub signin. Is this what we want?

---

The `validate_old_password` method from User is already taking that into account and returning true on purpose, so I decided to do the same... But yeah, I'm not sure what would be more reasonable for this API. Maybe it would be better to raise an exception in that case?

---

I like the exception idea as a third return option: 
- `true`: password is valid
- `false`: password is invalid
- Exception: User has no password set

I prefer that to the current behaviour of silently returning `true` to everything if the user has no password set. It could cause problems if the API consumers use it to verify passwords for login or similar purposes (it would login the user with any password).